projects/plugins/jetpack/extensions/blocks/recurring-payments/test/edit.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom/extend-expect';
+import { render, screen, waitFor } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { MembershipsButtonEdit } from '../edit';
+
+describe( 'MembershipsButtonEdit', () => {
+	const defaultAttributes = {
+		planId: null,
+	};
+
+	const setAttributes = jest.fn();
+	const updateBlockAttributes = jest.fn();
+	const onRemove = jest.fn();
+	const onReplace = jest.fn();
+	const autosaveAndRedirect = jest.fn();
+
+	const defaultProducts = [
+		{
+			id: 1,
+			currency: "USD",
+			price: "10.00",
+			interval: "1 month",
+			title: "ten a month",
+		},
+		{
+			id: 2,
+			currency: "EUR",
+			price: "5.00",
+			interval: "1 year",
+			title: "five a year",
+		}
+	];
+	const defaultProps = {
+		attributes: defaultAttributes,
+		setAttributes,
+		updateBlockAttributes,
+		onRemove,
+		onReplace,
+		autosaveAndRedirect,
+		clientId: 1,
+		postId: 1,
+		noticeList: [],
+		isSelected: true,
+		context: { },
+	};
+
+	const defaultApiResponse = {
+		connect_url: '',
+		connected_account_id: 1,
+		products: defaultProducts,
+		should_upgrade_to_access_memberships: false,
+		site_slug: 'test',
+		upgrade_url: 'https://wordpress.com/checkout/test/jetpack_security_daily_monthly',
+	};
+
+	jest.mock( '../../../shared/get-jetpack-extension-availability', () => () => ( {
+		available: false,
+		unavailableReason: 'missing_plan',
+	} ) );
+
+	// Mock out InnerBlocks.
+	jest.mock( '@wordpress/block-editor', () => ( {
+		...jest.requireActual( '@wordpress/block-editor' ),
+		InnerBlocks: () => ( '' ),
+	} ) );
+
```

Comment:
My idea here is to mock out the `InnerBlocks`, and test the bulk of the block's functionality (rendering nudges and managing payment plans), which are completely separate. Mocking the block out this way should be possible -- see docs: https://testing-library.com/docs/example-react-transition-group

I haven't been able to get this working. In the interim I'm just commenting out the `InnerBlocks` in the `edit.js` file while working through the other mocks.

Reply:
Resolved -- we also need to `registerBlocks( [ { name: 'jetpack/button', settings } ] );` within the test file 👍